src/HotChocolate/Core/src/Types/Types/Descriptors/Conventions/DefaultNamingConventions.cs

```diff
@@ -341,8 +341,17 @@ public NameString FormatFieldName(string fieldName)
                 nameof(fieldName));
         }
 
-        return fieldName.Length > 1
-            ? fieldName.Substring(0, 1).ToLowerInvariant() + fieldName.Substring(1)
-            : fieldName.ToLowerInvariant();
+        if(fieldName.Length < 2)
+            return fieldName.ToLowerInvariant();
+
+        var lastUpperCaseIndex = 0;
+
+        for (var i = 0; i < fieldName.Length; i++)
+        {
+            if (char.IsUpper(fieldName[i]))
+                lastUpperCaseIndex = i;
+        }
```

Comment:
I dont think this works.

Woudnt this also convert FooBarBaz to foobarBaz?

Reply:
tested it in a playground and it worked

Reply:
but yes you are right

Reply:
@Nivisan-Jeyananthan you can try something like this i guess (might be off by one ;D )
```csharp 
var lastUpperCaseIndex = -1; 

while(lastUperCaseIndex + 1 < fieldName.Lengh && 
     char.IsUpper(fieldName[lastUpserCaseIndex+1]))  
{
    lastUpperCaseIndex++;
}
if(lastUpperCaseIndex == -1) 
{
   return fieldName;
}
return fieldName.Substring(0, lastUpperCaseIndex).ToLowerInvariant() + 
  fieldName.Substring(lastUpperCaseIndex)

```

Reply:
@PascalSenn you mean you're a few characters short ;)